source/assets/js/playground/generate-module-members.ts

```diff
@@ -0,0 +1,92 @@
+// Gets list of all functions and variables for the built-in modules
+//
+// Outputs `./module-members.ts` file, which is omitted from source code and
+// must be built before the `playground` bundle is built with rollup.
+import {sassTrue, compileString} from 'sass';
+import {writeFileSync} from 'fs';
+import path from 'path';
+
+const modules = [
+  'color',
+  'list',
+  'map',
+  'math',
+  'meta',
+  'selector',
+  'string',
+] as const;
+
+type ModuleName = (typeof modules)[number];
+
+type ModuleDefinition = {
+  name: ModuleName;
+  functions?: string[];
+  variables?: string[];
+};
+
+// Generate Scss with a custom function that extracts each module's functions
+// and variables.
+function generateModuleMembers() {
+  // Generate Sass
+  const moduleSass = `
+${modules.map(mod => `@use "sass:${mod}";`).join('\n')}
+$modules: ${modules.join(',')};
+
+selector{
+  @each $module in $modules{
+    f: extract(map.keys(meta.module-functions($module)), $module, functions);
+    $variables: map.keys(meta.module-variables($module));
+    @if (list.length($variables) > 0){
+      v: extract(map.keys(meta.module-variables($module)), $module, variables);
+    }
+  }
+}
+`;
+
+  const modMap: ModuleDefinition[] = [];
```

Comment:
Would it be cleaner to make this a `Record<string, ModuleDefinition>` with the module names as keys? Then instead of some `Array.prototype.reduce` stuff, you could use a simpler `mapRecord<I, O>(record: Record<T, I>, mapper: (I) => O): Record<T, O>` utility function.

Reply:
Potentially it could be cleaner in the cases where I'm reducing things into `Record<ModuleName, CompletionSource>`, but the metadata is being used in other ways as well. I'm generally a fan of keeping data in flat arrays of objects, as it tends to be the most versatile.